websites/views.py

```diff
@@ -149,10 +150,19 @@ def get_serializer(self, *args, **kwargs):
     def preview(self, request, name=None):
         """Trigger a preview task for the website"""
         try:
-            preview_website(self.get_object())
+            website = self.get_object()
+            incomplete_videos = [
+                video.title for video in unassigned_youtube_ids(website)
+            ]
+            message = ""
+            if len(incomplete_videos) > 0:
+                message = f"WARNING: The following videos have missing YouTube IDs: {','.join(incomplete_videos)}"
+            # Try updating YouTube metadata first, because the preview should be aborted if it fails.
+            update_youtube_metadata(website)
```

Comment:
Can this be moved to settings.PREPUBLISH_ACTIONS or just to  content_sync.tasks.preview_website_backend?

I think we want to push the webvtt transcript to youtube with the rest of the metadata, and that's not available until the transcript fields are updated in content_sync.tasks.preview_website_backend. If you want we can merge it as is and I can make that change when I update transcripts on youtube.

Sorry! I know this and https://github.com/mitodl/ocw-studio/pull/552 were written at the same time

Reply:
Good point, I moved it to `settings.PREPUBLISH_ACTIONS` and added try-catch blocks so if anything goes wrong during preview/publish an email will be sent to the website admins.